x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/ml/apm_metrics.json

```diff
@@ -0,0 +1,53 @@
+{
+  "job_type": "anomaly_detector",
+  "groups": [
+    "apm"
+  ],
+  "description": "Detects anomalies in transaction duration, throughput and error percentage for metric data.",
+  "analysis_config": {
+    "bucket_span": "15m",
+    "summary_count_field_name" : "doc_count",
+    "detectors" : [
+      {
+        "detector_description" : "high duration by transaction type for an APM service",
+        "function" : "high_mean",
+        "field_name" : "transaction.duration.us",
```

Comment:
I am not 100% sure if we want these to be different field names than transaction.duration.histogram. It depends on how APM needs to interact with the resulting anomaly buckets/values. I am assuming that they are expecting the field to be called transaction.duration.histogram in their searches.

Reply:
Has this been resolved yet? When we first started investigating, we thought it might be useful to make the metric job results look exactly like the transaction job results, and therefore this would not need UI changes to display anomaly latency results. In order to do this, we named the result field to be `transation.duration.us` for both.  Things have changed since as the job now has additional detectors.

Going forwards, it is probably more useful for future maintainability to use `transaction.duration.histogram`. This avoids confusion as it will be obvious the data source is metric data.

Reply:
I agree on using `transaction.duration.histogram` as that pairs up the anomalous metric with the source data utilized. 

I don't know what APM thinks. 

@dgieselaar ^?

Reply:
quoting myself here:
> transaction.duration.us shows up in field_name as a value, not as an ES field name, correct?

I don't think we're using `transaction.duration.us` anywhere when interacting with ML data. All APM ML data would have been transaction duration anomaly data. In this PR however we have started using detector index to select the appropriate results.

To help my understanding, is `field_name` here actually referring to the aggregation name?

Reply:
`field_name` is indeed referring to the aggregation name. Additionally, the resulting anomaly `record` outputs will record the `field_name` which can help merging together the raw original data and our anomaly records.

Reply:
Yes, in the metric job config, `"field_name" : "transaction.duration.us"` is referring to the aggregation name defined in the datafeed. We should change this to be `transaction.duration.histogram` in both the datafeed and job configs. 

@peteharverson It seems this is not used anywhere else in code, I think it might be easier for you to make this change. OK?

Reply:
Is there a compelling reason to use the field name (`transaction.duration.histogram`)? Why not just `transaction_duration`? Ideally we use something like `transaction_failure_percentage` for `percent_failure` as well, to leave room for e.g. database call failures.

Reply:
Changed aggregation names in https://github.com/elastic/kibana/pull/111957/commits/ff43ca262b02a93724ab9ef8f09ca5f307b995d6 to the values suggested by @dgieselaar - `transaction_duration` and `transaction_failure_percentage`.